src/cargo/core/compiler/mod.rs

```diff
@@ -1162,6 +1168,69 @@ fn features_args(unit: &Unit) -> Vec<OsString> {
     args
 }
 
+/// Generates the `--remap-path-scope` and `--remap-path-prefix` for [RFC 3127].
+/// See also unstable feature [`-Ztrim-paths`].
+///
+/// [RFC 3127]: https://rust-lang.github.io/rfcs/3127-trim-paths.html
+/// [`-Ztrim-paths`]: https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#profile-trim-paths-option
+fn trim_paths_args(
+    cmd: &mut ProcessBuilder,
+    cx: &Context<'_, '_>,
+    unit: &Unit,
+    trim_paths: &TomlTrimPaths,
+) -> CargoResult<()> {
+    if trim_paths.is_none() {
+        return Ok(());
+    }
+
+    // feature gate was checked during mainfest/config parsing.
+    cmd.arg("-Zunstable-options");
+    cmd.arg(format!("-Zremap-path-scope={trim_paths}"));
+
+    let sysroot_remap = {
+        let sysroot = &cx.bcx.target_data.info(unit.kind).sysroot_target_libdir;
+        let commit_hash = &cx.bcx.rustc().commit_hash;
+        let mut remap = OsString::from("--remap-path-prefix=");
+        // TODO(trim-paths): what is the correct prefix for sysroot? does rustup affect this?
+        remap.push(sysroot);
```

Comment:
> What's the correct remap prefix for sysroot

[`rustc --print=sysroot`](https://doc.rust-lang.org/rustc/command-line-arguments.html#--print-print-compiler-information)

> and how to setup a test case for it?

Well you could have a Rust `fn main() { panic!("something"); }` with `RUST_BACKTRACE=1` and see the resulting backtrace and assert that it doesn't contain the un-remapped sysroot.

Reply:
Thanks for the tip!

Unfortunately I've tried `panic!` backtrace and it has already remapped even without `-Ztrim-paths` 😞.

```
   0: std::panicking::begin_panic
             at /rustc/cd674d61790607dfb6faa9d754bd3adfa13aea7c/library/std/src/panicking.rs:638:12
```

Reply:
The desired behavior is mentioned in the RFC

* https://rust-lang.github.io/rfcs/3127-trim-paths.html#handling-sysroot-paths
* https://rust-lang.github.io/rfcs/3127-trim-paths.html#changing-handling-of-sysroot-path-in-rustc

To follow the RFC, yes, we might want to disable it in bootstrap.

Reply:
Me and Urgau had a chat on this. Urgau pointed out changing the bootstrap this may require `-Ztrim-paths` being stabilized, and may leak CI details into all pre-built std, core…. This is something requiring to collaborate with T-bootstrap. We'll postpone it for this pull request.